trk/src/Screens/GymDaily/index.js

```diff
@@ -1,8 +1,219 @@
+
+      if (mostCompletedClimb) {
+        setMostCompleted({ name: mostCompletedClimb.name, id: mostCompletedClimb.id, grade: mostCompletedClimb.grade });
+      } else {
+        setMostCompleted({ name: '', id: '', grade: '' });
+      }
+
+      if (leastCompletedClimb) {
+        setLeastCompleted({ name: leastCompletedClimb.name, id: leastCompletedClimb.id, grade: leastCompletedClimb.grade });
+      } else {
+        setLeastCompleted({ name: '', id: '', grade: '' });
+      }
+
+    };
+
+
+    if (yourClimbs.length > 0) {
+      fetchTapsAndCalculateAscents();
+    }
+  }, [yourClimbs]);
+
+
+  useEffect(() => {
+    const fetchCommentsAndCalculateFeedback = async () => {
+      try {
+        const commentsPromises = yourClimbs.map(climb =>
+          getCommentsBySomeField('climb', climb.id)
+        );
+        const commentsSnapshots = await Promise.all(commentsPromises);
+        let comments = [];
+        commentsSnapshots.forEach(snapshot => {
+          snapshot.docs.forEach(doc => comments.push({ ...doc.data(), timestamp: doc.data().timestamp.toDate() }));
+        });
+
+        setYourComments(comments);
+ 
+        if (comments.length > 0) {
+          const latestComment = comments.reduce((latest, current) =>
+            latest.timestamp > current.timestamp ? latest : current
+          );
+
+          const climbDetails = await getClimb(latestComment.climb);
+          const climbData = climbDetails.data();
+          setLatestFeedback({
+            explanation: latestComment.explanation,
+            climb: latestComment.climb,
+            rating: latestComment.rating,
+            climbName: climbData.name,
+            climbGrade: climbData.grade,
+          });
+        }
+
+        let climbRatings = {};
+        comments.forEach(comment => {
+          if (!climbRatings[comment.climb]) {
+            climbRatings[comment.climb] = { totalRating: 0, count: 0 };
+          }
+          climbRatings[comment.climb].totalRating += comment.rating;
+          climbRatings[comment.climb].count++;
+        });
+
+        let highestRatedClimbId = Object.keys(climbRatings).reduce((a, b) =>
+          (climbRatings[a].totalRating / climbRatings[a].count) > (climbRatings[b].totalRating / climbRatings[b].count) ? a : b
+        );
+
+        const highestRatedClimbDetails = await getClimb(highestRatedClimbId);
+        const climbData = highestRatedClimbDetails.data();
+        setHighestRated({
+          name: climbData.name,
+          grade: climbData.grade,
+          type: climbData.type,
+          climb: highestRatedClimbId,
+        });
+
+
+      } catch (error) {
+        console.error("Error fetching comments:", error);
+      }
+    };
+
+    if (yourClimbs.length > 0) {
+      fetchCommentsAndCalculateFeedback();
+    }
+  }, [yourClimbs]);
+
+  const getStars = (rating) => {
+    let starRating = '';
+    for (let i = 0; i < rating; i++) {
+      starRating += '⭐️';
+    }
+    return starRating;
+  };
+
+
```

Comment:
convert the rating to a number of stars (note that rating is either 1,2,3,4,5 and cannot be a decimal value)